cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -599,8 +648,22 @@ public QueryHandle execute(String statement) throws ExploreException, SQLExcepti
       // It looks like the username and password below is not used when security is disabled in Hive Server2.
       SessionHandle sessionHandle = cliService.openSession("", "", sessionConf);
       try {
+        // Switch database to the one being passed in.
+        final OperationHandle useDatabaseHandle = doExecute(sessionHandle, String.format("USE %s", namespace));
```

Comment:
Also, check what happens when an invalid namespace is given. Does the error message look meaningful? If not should we validate the namespace before using it?